open_policy_agent/datadog_checks/open_policy_agent/open_policy_agent.py

```diff
@@ -0,0 +1,69 @@
+# (C) Datadog, Inc. 2020
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheck
+
+from .metrics import METRIC_MAP
+
+
+class OpenPolicyAgentCheck(OpenMetricsBaseCheck):
+    DEFAULT_METRIC_LIMIT = 0
+
+    def __init__(self, name, init_config, instances=None):
+        default_instances = {
+            'open_policy_agent': {
+                'metrics': [METRIC_MAP],
+                'send_distribution_sums_as_monotonic': 'true',
+                'send_distribution_counts_as_monotonic': 'true',
+            }
+        }
+
+        super(OpenPolicyAgentCheck, self).__init__(
+            name, init_config, instances, default_instances=default_instances, default_namespace='open_policy_agent'
+        )
+
+    def _http_check(self, url, check_name):
+        try:
+            response = self.http.get(url)
+            response.raise_for_status()
+        except Exception as e:
+            self.service_check(check_name, self.CRITICAL, message=str(e))
+        else:
+            if response.status_code == 200:
+                self.service_check(check_name, self.OK)
+            else:
+                self.service_check(check_name, self.WARNING)
+
+    def _get_policies(self, opa_url):
+        policies_url = opa_url + "/v1/policies"
+        try:
+            response = self.http.get(policies_url)
+            policies = response.json()
+        except Exception:
+            self.log.warning("The policies endpoint is not available")
+        else:
+            self.gauge('open_policy_agent.policies', len(policies['result']), tags=[])
+
+    def check(self, instance):
+        opa_url = instance.get("opa_url")
```

Comment:
```suggestion
    def check(self, _):
        opa_url = self.instance.get("opa_url")
```
(nit) In the futur, `check(self, instance)` will be deprecated. It's not a blocker, but it's a better practice to start using `self.instance` instead of the instance argument of the check method